ui/src/PartCommands.cpp

```
@@ -0,0 +1,696 @@
+                               SelectionSignature::exactly(EntityKind::Sketch, 1));
+    c.schema.push_back(ParamSpec{"distance", ParamType::Number, true, 10.0, ""});
+    c.schema.push_back(ParamSpec{"dirx", ParamType::Number, false, 0.0, ""});
+    c.schema.push_back(ParamSpec{"diry", ParamType::Number, false, 0.0, ""});
+    c.schema.push_back(ParamSpec{"dirz", ParamType::Number, false, 1.0, ""});
+    c.preview = PreviewPolicy::Live;
+    c.enabled = [d](const CommandContext& ctx) {
+      return resolveValues(*d, ctx.selection(), IrValueKind::Profile).size() == 1 &&
+             num(ctx, "distance", 0.0) != 0.0;  // a zero-height extrude is not a solid
+    };
+    c.execute = [d, s](CommandContext& ctx) {
+      const int profile = resolveValues(*d, ctx.selection(), IrValueKind::Profile).front();
```

Clang (14.0.6)

[warning] variable 'profile' is not initialized

(cppcoreguidelines-init-variables)